src/MIDebugEngine/Natvis.Impl/Natvis.cs

```diff
@@ -321,6 +321,20 @@ public void Initialize(string fileName)
             }
         }
 
+        /*
+         * Handle multiple Natvis files
+         */
+        public void Initialize(List <string> fileNames)
+        {
+            if (fileNames.Count > 0)
+            {
+                foreach (var file in fileNames)
+                {
+                    Initialize(file);
```

Comment:
>Initialize(file); [](http://example.com/codeflow?start=20&length=17)

Not all of Initialize should be repeated, and parts of Initialize look like they should be run even if there are zero files.

Reply:
> Not all of Initialize should be repeated,

Yep. There does not need to be multiple registrations of the natvis registration key. 

>and parts of Initialize look like they should be run even if there are zero files.

Everything else seems to be under if (!string.IsNullOrEmpty(fileName))

Reply:
I believe you will want to move the following to its own method and call it if `List<string> filenames` is not empty,
```
void InitializeNatvisServices() {
            try
            {
                _natvisSettingWatcher = HostNatvisProject.WatchNatvisOptionSetting(_configStore, _process.Logger.NatvisLogger);
                HostNatvisProject.FindNatvis((s) => LoadFile(s));
            }
            catch (FileNotFoundException)
            {
                // failed to find the VS Service
            }
}
```

Call this method in the list<string> initialize function.